gtsam/hybrid/IncrementalHybrid.cpp

```diff
@@ -140,23 +142,34 @@ void IncrementalHybrid::update(GaussianHybridFactorGraph graph,
         discreteFactor->enumerate();
 
     // Loop over all assignments and create a vector of GaussianConditionals
-    std::vector<GaussianFactor::shared_ptr> prunedConditionals;
-    for (auto &&av : assignments) {
-      const DiscreteValues &assignment = av.first;
-      const double value = av.second;
-
-      if (value == 0.0) {
-        prunedConditionals.emplace_back(nullptr);
-      } else {
-        prunedConditionals.emplace_back(lastDensity->operator()(assignment));
+    std::vector<GaussianMixture::shared_ptr> lastClique;
```

Comment:
I don't understand why looping over an extra conditional is taking care of the pruning problem.
Can you please elaborate in detail?